docs/source/execute_api.rst

```diff
@@ -0,0 +1,145 @@
+allowed to run, if the execution takes longer an exception will be raised.
+The default is only 30 s, so in many cases you may want to specify
+an higher value.
+
+The second traitlet, ``kernel_name``, allows specifying the name of the kernel
+to be used for the execution. By default, the kernel name is obtained from the
+notebook metadata. The traitlet ``kernel_name`` allows to specify a user-defined
+kernel, overriding the value in the notebook metadata. A common use case
+is that of a python 2/3 library which includes documentation/testing
+notebooks. These notebooks will specify either a python2 or python3 kernel
+in their metadata
+(depending on the kernel used the last time the notebook was saved).
+In reality, these notebooks will work on both python 2/3 and, for testing,
+it is important to be able to execute them programmatically on both
+versions. Here the traitlet ``kernel_name`` comes to help:
+we can just run each notebook twice, specifying first "python2" and then
+"python3" as kernel name.
+
+Error Handling
+--------------
+
+In the previous sections we saw how to save an executed notebook, assuming
+there are no execution error. But, what if there are errors?
+
+An error during the notebook execution, by default, will stop the execution
+and raise a ``CellExecutionError``. Conveniently, the source cell causing
+the error and the original error name and message are also printed.
+After an error, we can still save the notebook as before:
+
+.. code-block:: python
+
+    nbformat.write(nb, open('executed_notebook.ipynb', mode='wt'))
+
+The saved notebook contains the output up until the failing cell,
+and includes a full stack-trace and error (which can help debugging).
+A pattern I use to execute notebooks while handling errors is the following:
+
+.. code-block:: python
+
+    try:
+        out = ep.preprocess(nb, {'metadata': {'path': run_path}})
+    except CellExecutionError:
+        msg = 'Error executing the notebook "%s".\n\n' % notebook_filename
+        msg += 'See notebook "%s" for the traceback.' % notebook_filename_out
+        print(msg)
+        raise
+    finally:
+        nbformat.write(nb, open(notebook_filename_out, mode='wt'))
+
+This will save the executed notebook regardless of execution errors.
+In case of errors, however, an additional message is printed and the
+``CellExecutionError`` is raised. The messages directs the user to
+the saved notebook for further inspection.
+
+As a last scenario, sometimes notebooks contains independent computations
+in each code cell.
+In this case it can be useful to run the notebook until the end,
+in order to get a complete picture of all cells that are failing.
+Luckily enough, the ``allow_errors`` traitlet (default False) allows to do that.
+With ``allow_errors=True``,
+the notebook is executed until the end, and a ``CellExecutionError`` is raised
+if one or more cells threw an error. In this case, the output notebook
```

Comment:
Ops, yes, in fact due to the switched True/False I mis-interpreted it as raising an exception (but at the end) when `allow_errors=True`. I'm including a docstring fix in the updated PR.